### PR TITLE
Value tree node

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: spec/augeas_parser_spec.rb [hunk 1]
```diff
@@ -10,6 +10,13 @@
       expect(subject.parse(example_file)).to be_a(CFA::AugeasTree)
     end
 
+    it "can handle augeas with value and three below" do
```
Comment (lslezak) [inline spec/augeas_parser_spec.rb]: Typo `tree`, plus `and a tree below` better would be even better...

File: CHANGELOG [hunk 2]
```diff
@@ -1,4 +1,11 @@
 -------------------------------------------------------------------
+Tue Sep 27 09:17:51 UTC 2016 - jreidinger@suse.com
+
+- support augeas nodes containing value and also attached tree
+  below it like e.g. ntp.conf have (bnc#983486)
```
Comment (lslezak) [inline CHANGELOG]: `like ntp.conf has`

File: lib/cfa/augeas_parser.rb [hunk 3]
```diff
@@ -70,13 +88,17 @@ def add(key, value, placer = AppendPlacer.new)
       element[:value] = value
     end
 
+    # finds given value in tree. It can return value of node, {AugeasTree}
+    # attached to key or its combination as {AugeasTreeValue}
```
Comment (lslezak) [inline lib/cfa/augeas_parser.rb]: ... or `nil` if not found.

File: lib/cfa/augeas_parser.rb [hunk 4]
```diff
@@ -70,13 +88,17 @@ def add(key, value, placer = AppendPlacer.new)
...
     def [](key)
       entry = @data.find { |d| d[:key] == key }
       return entry[:value] if entry
 
       nil
     end
 
+    # sets to given key value in tree. It can be value of node, {AugeasTree}
```
Comment (lslezak) [inline lib/cfa/augeas_parser.rb]: `Sets the given value for the key in tree`

File: lib/cfa/augeas_parser.rb (hunk 3 above)
Comment (lslezak) [inline lib/cfa/augeas_parser.rb]: Maybe use the `@return` comment?

File: lib/cfa/augeas_parser.rb [hunk 5]
```diff
@@ -112,33 +142,25 @@ def save_to_augeas(aug, prefix)
...
       end
     end
 
-  private
-
-    def obtain_aug_key(prefix, entry, arrays)
-      key = entry[:key]
+    def obtain_aug_key(prefix, key, arrays)
       if key.end_with?("[]")
         array_key = key.sub(/\[\]$/, "")
```
Comment (lslezak) [inline lib/cfa/augeas_parser.rb]: NP: This looks too complicated, regexp (with escapes) substitution with empty string....

What about simple `key.chomp("[]")` ? (See http://ruby-doc.org/core-2.3.1/String.html#method-i-chomp)

File: lib/cfa/augeas_parser.rb [hunk 6]
```diff
@@ -228,13 +253,13 @@ def empty
     def report_error(aug)
       error = aug.error
       # zero is no error, so problem in lense
-      if aug.error[:code] != 0
+      if aug.error[:code].nonzero?
```
Comment (lslezak) [inline lib/cfa/augeas_parser.rb]: This does not handle `nil` as the original code, is a non-nil guaranteed?

File: lib/cfa/augeas_parser.rb (hunk 5 above)
Comment (jreidinger) [inline lib/cfa/augeas_parser.rb]: or maybe more efficient `[0..-3]`?

File: lib/cfa/augeas_parser.rb (hunk 6 above)
Comment (jreidinger) [inline lib/cfa/augeas_parser.rb]: it is defined value of error

Comment (lslezak) [inline lib/cfa/augeas_parser.rb]: I see, then it's OK...

**Pull request merged**